File: vibe-lang/benchmark/output/lua-love/openai/pong_game.lua
local p1_y = 260.0
local p2_y = 260.0
local ball_x = 400.0
local ball_y = 300.0
local ball_vx = 250.0
local ball_vy = 150.0
local score1 = 0
local score2 = 0
local paddle_speed = 300.0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("2-Player Pong")
end

function love.update(dt)
    -- Move left paddle
    if love.keyboard.isDown("w") then
        p1_y = math.max(0, p1_y - paddle_speed * dt)
    end
    if love.keyboard.isDown("s") then
        p1_y = math.min(520, p1_y + paddle_speed * dt)
    end

    -- Move right paddle
    if love.keyboard.isDown("up") then
        p2_y = math.max(0, p2_y - paddle_speed * dt)
    end
    if love.keyboard.isDown("down") then
        p2_y = math.min(520, p2_y + paddle_speed * dt)
    end

    -- Move ball
    ball_x = ball_x + ball_vx * dt
    ball_y = ball_y + ball_vy * dt

    -- Ball collision with top and bottom
    if ball_y < 8 then
        ball_vy = -ball_vy
        ball_y = 8
    elseif ball_y > 592 then
        ball_vy = -ball_vy
        ball_y = 592
    end

    -- Ball collision with left paddle
    if ball_x - 8 <= 42 and ball_x > 30 and ball_y >= p1_y and ball_y <= p1_y + 80 then
        ball_vx = -ball_vx
        ball_x = 50
    end

    -- Ball collision with right paddle
    if ball_x + 8 >= 758 and ball_x < 770 and ball_y >= p2_y and ball_y <= p2_y + 80 then
        ball_vx = -ball_vx
        ball_x = 750
    end

    -- Ball out of bounds
    if ball_x < 0 then
        score2 = score2 + 1
        ball_x = 400
        ball_y = 300
        ball_vx = 250
        ball_vy = 150
    elseif ball_x > 800 then
        score1 = score1 + 1
        ball_x = 400
        ball_y = 300
        ball_vx = -250
        ball_vy = -150
    end
end

function love.draw()
    -- Draw paddles
    love.graphics.rectangle("fill", 30, p1_y, 12, 80)
    love.graphics.rectangle("fill", 758, p2_y, 12, 80)

    -- Draw ball
    love.graphics.circle("fill", ball_x, ball_y, 8)

    -- Draw center line
    for i = 0, 29 do
        love.graphics.rectangle("fill", 398, i * 20, 4, 10)
    end

    -- Draw scores
    love.graphics.print(tostring(score1), 300, 20)
    love.graphics.print(tostring(score2), 480, 20)
end

File: benchmark/output/python-pygame/openai/pong_game.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("2-Player Pong")
clock = pygame.time.Clock()

p1_y = 260.0
p2_y = 260.0
ball_x = 400.0
ball_y = 300.0
ball_vx = 250.0
ball_vy = 150.0
score1 = 0
score2 = 0
paddle_speed = 300

def update(dt):
    global p1_y, p2_y, ball_x, ball_y, ball_vx, ball_vy, score1, score2

    keys = pygame.key.get_pressed()
    if keys[pygame.K_w]:
        p1_y -= paddle_speed * dt
    if keys[pygame.K_s]:
        p1_y += paddle_speed * dt
    if keys[pygame.K_UP]:
        p2_y -= paddle_speed * dt
    if keys[pygame.K_DOWN]:
        p2_y += paddle_speed * dt

    p1_y = max(0.0, min(520.0, p1_y))
    p2_y = max(0.0, min(520.0, p2_y))

    ball_x += ball_vx * dt
    ball_y += ball_vy * dt

    if ball_y < 8.0 or ball_y > 592.0:
        ball_vy = -ball_vy

    if ball_x - 8.0 <= 42.0 and ball_x > 30.0 and p1_y <= ball_y <= p1_y + 80.0:
        ball_vx = -ball_vx
        ball_x = 50.0

    if ball_x + 8.0 >= 758.0 and ball_x < 770.0 and p2_y <= ball_y <= p2_y + 80.0:
        ball_vx = -ball_vx
        ball_x = 750.0

    if ball_x < 0.0:
        score2 += 1
        ball_x, ball_y = 400.0, 300.0
        ball_vx, ball_vy = 250.0, 150.0

    if ball_x > 800.0:
        score1 += 1
        ball_x, ball_y = 400.0, 300.0
        ball_vx, ball_vy = -250.0, -150.0

def draw():
    screen.fill((0, 0, 0))
    pygame.draw.rect(screen, (255, 255, 255), (30, p1_y, 12, 80))
    pygame.draw.rect(screen, (255, 255, 255), (758, p2_y, 12, 80))
    pygame.draw.circle(screen, (255, 255, 255), (int(ball_x), int(ball_y)), 8)

    for i in range(0, 30):
        pygame.draw.rect(screen, (255, 255, 255), (398, i * 20, 4, 10))

    font = pygame.font.Font(None, 36)
    text1 = font.render(str(score1), True, (255, 255, 255))
    text2 = font.render(str(score2), True, (255, 255, 255))
    screen.blit(text1, (300, 20))
    screen.blit(text2, (480, 20))

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    update(dt)
    draw()
    pygame.display.flip()

pygame.quit()
sys.exit()